addons/mail/models/mail_thread.py

```diff
@@ -2397,6 +2398,10 @@ def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
             return True
         force_send = self.env.context.get('mail_notify_force_send', force_send)
 
+        if scheduled_date:
```

Comment:
If we add the field on mail.message, we enable its usage in templates, and we won't have to add a specific parameter (would simply have a ``scheduled_date = msg_vals.get('scheduled_date') if msg_vals and 'scheduled_date' in msg_vales else message.scheduled_date``). That way we have a clean support of it at model level.